index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
What is wrong with `when :: (a -> a) -> a -> a`, @cust0dian? I see no problem with implementing this in terms of `ifElse`, since there's no requirement that `a` and `b` be replaced with *different* types.